File: backend/src/main/java/com/mafia/model/enums/GamePhase.java
package com.mafia.model.enums;

/**
 * Represents the current phase of the game.
 */
public enum GamePhase {
    WAITING,         // Lobby created, waiting for players
    ROLES_ASSIGNED,  // God has assigned roles, ready to start
    NIGHT,           // Night phase (Mafia acts)
    DAY,             // Day phase (discussion and voting)
    FINISHED         // Game ended
}

File: frontend/src/types/GamePhase.ts
/**
 * Game Phase Type
 *
 * Keep in sync with backend: com.mafia.model.enums.GamePhase
 */

export type GamePhase =
  | "WAITING"
  | "ROLES_ASSIGNED"
  | "NIGHT"
  | "DAY"
  | "FINISHED";
